### PR TITLE
RichText: Fix outdated docs

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

* Includes changing some `children` references.
* Removes suggesting the use of the `format` prop because all that should be used now is `string`.
* Adjusts some other outdated things.

### CODE REVIEW

**Review by tofumatt — APPROVED** [review]
Works for me, just left a comment about the `value` explanation.

**Comment by gziolo** [comment]
We don't need to cherry-pick it to 4.0. 4.1 is fine.

File: packages/editor/src/components/rich-text/README.md [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
-Render a rich [`contenteditable` input](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Editable_content), providing users the option to add emphasis to content or links to content. It behaves similarly to a [controlled component](https://facebook.github.io/react/docs/forms.html#controlled-components), except that `onChange` is triggered less frequently than would be expected from a traditional `input` field, usually when the user exits the field.
+Render a rich [`contenteditable` input](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Editable_content), providing users with the option to format the content.
 
 ## Properties
 
-### `format: String`
+### `value: String`
 
-*Optional.* Format of the RichText provided value prop. It can be `children` or `string`.
+*Required.* HTML string to make editable. The rendered HTML should be valid, and valid with respect to the `tagName` and `multiline` property.
```
[the remark this answers is not in the document]
Comment (iseulde) [reply packages/editor/src/components/rich-text/README.md]: I think it more means that you should not use any tags that cannot be children of the `tagName` you provide. The example you're giving might confuse people in that they'd have to use multiline in combination with `tagName="p"`? Maybe we can just leave this away.

[the remark this answers is not in the document]
Comment (iseulde) [reply packages/editor/src/components/rich-text/README.md]:
```suggestion
*Required.* HTML string to make editable. The HTML should be valid, and valid inside the `tagName`, if provided.
```

**Review by gziolo — APPROVED** [review]
Thanks for making it more precise 👍

**Pull request merged**